Don't overwrite processingData slot in initialize,MSnSet; fixes #264

sgibb opened the pull request:
This should be a trivial and non-invasive change to fix #264. I didn't merge it directly into master because:

1. I am not sure if you want a change an `initialize` method a few days before the next bioc release.
2. I got 49 failed unit tests. IMHO they are not related to this PR but have something to do with `OnDiskMSnExp`:

```
1. Error: addIdentificationData to OnDiskMSnExp (@test_MSnExp.R#224) -----------
"Multi_Spectrum2_constructor_mz_sorted" not available for .Call() for package "MSnbase"
...
49. Error: readMSData inMemory and onDisk reading CDF (@test_readMSData2.R#76) -
"Multi_Spectrum1_constructor_mz_sorted" not available for .Call() for package "MSnbase"
```

@lgatto and @jotsetung Do you have an idea why this `.Call` is not available? Do you need more information (`R CMD check` fails with the same error: [00check.log.txt](https://github.com/lgatto/MSnbase/files/1353905/00check.log.txt), [sessionInfo.txt](https://github.com/lgatto/MSnbase/files/1353903/sessionInfo.txt), [testthat.Rout.fail.txt](https://github.com/lgatto/MSnbase/files/1353904/testthat.Rout.fail.txt))?

jotsetung commented:
@sgibb , I don't get these errors (but another one not related to `OnDiskMSnExp` (see below). I guess it might have to do with the R-version you are using. Could it be that R-devel is already the upcoming 3.5? I'm still running R-3.4.2.

```r
* checking examples ... ERROR
Running examples in ‘MSnbase-Ex.R’ failed
The error most likely occurred in:

> ### Name: FeatComp-class
> ### Title: Class '"FeatComp"'
> ### Aliases: FeatComp-class compfnames-methods
> ###   compfnames,MSnSet,MSnSet-method compfnames,list,missing-method
> ###   compfnames show,FeatComp-method names,FeatComp-method
> ###   common,FeatComp-method common,methods common unique1,FeatComp-method
> ###   unique1,methods unique1 unique2,FeatComp-method unique2,methods
> ###   unique2
> ### Keywords: classes
> 
> ### ** Examples
> 
> library("pRolocdata")

This is pRolocdata version 1.15.0.
Use 'pRolocdata()' to list available data sets.
> data(tan2009r1)
> data(tan2009r2)
> x <- compfnames(tan2009r1, tan2009r2)
> x[[1]]
Object of class "FeatComp", 'all' features:
 Common feature: 551 
 Unique to 1: 337 
 Unique to 2: 320 
> x[2:3]
[[1]]
Object of class "FeatComp", 'unknown' features:
 Common feature: 413 
 Unique to 1: 264 
 Unique to 2: 331 

[[2]]
Object of class "FeatComp", 'ER' features:
 Common feature: 21 
 Unique to 1: 7 
 Unique to 2: 0 

> head(common(x[[1]]))
[1] "P20353" "P53501" "Q7KU78" "P04412" "Q7JZN0" "Q9VM65"
> 
> data(tan2009r3)
> tanl <- list(tan2009r1, tan2009r2, tan2009r3)
> xx <- compfnames(tanl, fcol1 = NULL)
Error in names(ans) <- apply(cmbs, 2, paste, collapse = "v") : 
  'names' attribute [30] must be the same length as the vector [15]
Calls: compfnames -> compfnames
Execution halted
```

lgatto commented:
Works for me with 3.4.0 Patched (2017-06-12 r72786). I'll update R to 3.4.2 and try again later today.

lgatto commented:
Looks good with latest patched version - merging!

sgibb commented:
@jotsetung I removed the whole R directory, updated to the most recent R-devel from svn and reinstalled all necessary packages and now it runs without an error. Sorry for the noise. I have no idea what was wrong.
